packages/plugin-line-spacing/src/toolbar/LineSpacingButton.jsx

```diff
@@ -0,0 +1,154 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import pick from 'lodash/pick';
+import {
+  getAnchorBlockData,
+  getModalStyles,
+  InlineToolbarButton,
+  mergeBlockData,
+  mergeStyles,
+} from 'wix-rich-content-common';
+
+import { LineSpacingIcon } from '../icons';
+import styles from '../../statics/styles/styles.scss';
+import { MODAL_STYLES } from './modalStyles';
+import { Modals } from '../modals';
+
+const lineHeight = 'line-height';
+const spaceBefore = 'padding-top';
+const spaceAfter = 'padding-bottom';
+
+export default class LineSpacingButton extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { showPanel: false };
+    this.styles = mergeStyles({ styles, theme: props.theme });
+  }
+
+  getBlockSpacing() {
+    const { dynamicStyles = {} } = getAnchorBlockData(this.props.getEditorState());
+    return pick(dynamicStyles, [lineHeight, spaceBefore, spaceAfter]);
+  }
+
+  updateSpacing = spacing => {
+    const dynamicStyles = spacing;
+    const { setEditorState, onUpdate } = this.props;
+    const newEditorState = mergeBlockData(this.oldEditorState, { dynamicStyles }); //using EditorState with selection fixes 'undo'.
+    setEditorState(newEditorState);
+    onUpdate(dynamicStyles);
+  };
+
+  openPanel = () => {
+    const { setKeepOpen, isMobile, helpers, t, defaultSpacing, keyName } = this.props;
+    setKeepOpen(true);
+    this.oldEditorState = this.props.getEditorState();
+
+    const spacing = this.getBlockSpacing();
+    this.oldSpacing = spacing;
+    this.setState({ isPanelOpen: true });
+
+    if (helpers && helpers.openModal) {
+      this.props.setKeepOpen(true);
```

Comment:
you call `setKeepOpen(true)` in line 43 as well, are they both necessary?
```suggestion
      setKeepOpen(true);
```